### PR TITLE
Add KKP addon for KubeFlow operator

### PR DESCRIPTION
**Pull request by jiachengxu**

### CODE REVIEW

File: addon/kubeflow-operator/manifest.yaml [hunk 1]
```diff
@@ -0,0 +1,472 @@
...
+{{ if eq .Variables.IstioIngressGatewayServiceType "LoadBalancer" }}
+      - name: serviceType
+        value: LoadBalancer
+{{ else if eq .Variables.IstioIngressGatewayServiceType "NodePort" }}
+      - name: serviceType
+        value: NodePort
+{{ else if eq .Variables.IstioIngressGatewayServiceType "ClusterIP" }}
+      - name: serviceType
+        value: ClusterIP
+{{ end }}
```
Comment (jiachengxu) [inline addon/kubeflow-operator/manifest.yaml:151]: We pass service type to kubeflow operator

File: addon/kubeflow-operator/manifest.yaml [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+              fieldPath: metadata.namespace
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: OPERATOR_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        image: quay.io/kubermatic/kubeflow-operator:latest
```
Comment (jiachengxu) [inline addon/kubeflow-operator/manifest.yaml:116]: Here we are using our own operator image.

File: addon/addon.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apiVersion: kubermatic.k8s.io/v1
```
Comment (rastislavs) [inline addon/addon.yaml]: Maybe this file should rather go into the examples/ folder? As it is not part of the addon itself.

Comment (jiachengxu) [reply addon/addon.yaml]: yep, you are right, will move it

File: addon/kubeflow-operator/manifest.yaml [hunk 4]
```diff
@@ -0,0 +1,472 @@
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (jiachengxu) [inline addon/kubeflow-operator/manifest.yaml:6]: @rastislavs I was thinking about splitting the entire manifest.yaml file into 2 files, one for kubeflow operator, and one for KfDef, is there a way for us to make sure that addon controller can first apply the operator manifest and then the KfDef?

File: addon/addon.yaml (hunk 3 above)
Comment (jiachengxu) [reply addon/addon.yaml]: Done, also I have tested with dev cluster, and it works.

File: addon/kubeflow-operator/manifest.yaml (hunk 4 above)
Comment (rastislavs) [reply addon/kubeflow-operator/manifest.yaml:6]: God idea... It seems that reading the manifests goes down to this func: https://github.com/kubermatic/kubermatic/blob/master/pkg/addon/template.go#L187

"ReadDir reads the directory named by dirname and returns a list of directory entries sorted by filename."

So we could control the order by conscious naming :)

Comment (jiachengxu) [reply addon/kubeflow-operator/manifest.yaml:6]: Nice to know that :)
How do we keep the order and find meaningful names at the same time?
I was thinking about something like:
```
- operator.yaml
- kfdef.yaml
```
but the order is wrong...
Also, I am not sure if we should depend on the order in addon implementation, I just don't know if they will change anything in the future, which will break our addon :)

Comment (rastislavs) [reply addon/kubeflow-operator/manifest.yaml:6]: Right, let's keep it like it is for now.. Maybe we could solve this common problem with CRDs on the addon controller level first.

Comment (jiachengxu) [reply addon/kubeflow-operator/manifest.yaml:6]: Agree :)

**Pull request merged**